RecoLocalTracker/SiPixelRecHits/interface/SiPixelTemplate2D.h

```diff
@@ -76,7 +100,11 @@ struct SiPixelTemplateHeader2D {           //!< template header structure
 
 struct SiPixelTemplateStore2D { //!< template storage structure
    SiPixelTemplateHeader2D head;
-   boost::multi_array<SiPixelTemplateEntry2D,2> entry;     //!< use 2d entry to store [47][5] barrel entries or [5][9] fpix entries
+#ifndef SI_PIXEL_TEMPLATE_USE_BOOST
+   SiPixelTemplateEntry2D entry[73][7];  //!< use 2d entry to store [47][5] barrel entries or [5][9] fpix
```

Comment:
hi @makortel - here is one ( a single instances is 88kB)

Reply:
Thanks. I presume `SI_PIXEL_TEMPLATE_USE_BOOST` is not defined in CMSSW by default (or at least I didn't find any indications it being defined).

How did you get 88 kB? I got for the size of SiPixelTemplateEntry2D 29072 B, so multiplying that by 73*7 gives ~14 MB.

`SiPixelTemplateStore2D` looks to be first allocated (on stack!) here
https://github.com/cms-sw/cmssw/blob/a4d0c8a01163417f6cfd755b12a508bd5676d8f1/RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate2D.cc#L270
which is then probably enough to blow up the stack at the beginning of `SiPixelTemplate2D::pushfile()` function.

Reply:
I guess the simplest (not necessarily the best) fix would be along
```cpp
--- a/RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate2D.cc
+++ b/RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate2D.cc
@@ -267,7 +267,8 @@ bool SiPixelTemplate2D::pushfile(const SiPixel2DTemplateDBObject& dbobject, std:
    SiPixel2DTemplateDBObject db = dbobject;
    
    // Create a local template storage entry
-   SiPixelTemplateStore2D theCurrentTemp;
+   auto tmpPtr = std::make_unique<SiPixelTemplateStore2D>(); // must be allocated on the heap because of being too large (14 MB) for stack
+   auto& theCurrentTemp = *tmpPtr;
    
    // Fill the template storage for each template calibration stored in the db
    for(int m=0; m<db.numOfTempl(); ++m)
```
I tested it and the job runs fine.

Reply:
Hi,

you already pointed to the lines I meant (sorry for my current unavailability, I'm at a conference this week).

Addition from my side:
With the implementation of the 73x7 array I actually did not get very happy, since we now have different sizes of templates that we need to be abled to read in. Therefore I indeed used the BOOST option, which then does a resizing of the array.

If there's another, more preferrable option to boost, I'd be happy to give it a try/implement it.